great_expectations/data_context/util.py

```diff
@@ -198,6 +203,21 @@ def mask_db_url(cls, url: str, use_urlparse: bool = False, **kwargs) -> str:
             )
         return cls._mask_db_url_no_sa(url=url)
 
+    @classmethod
+    def _obfuscate_azure_blobstore_connection_string(cls, url: str) -> str | None:
+        # Parse Azure Connection Strings
+        azure_conn_str_re = re.compile(
+            "(DefaultEndpointsProtocol=(http|https));(AccountName=([a-zA-Z0-9]+));(AccountKey=)(.+);(EndpointSuffix=([a-zA-Z\\.]+))"
+        )
+        matched: re.Match[str] | None = azure_conn_str_re.match(url)
+        try:
+            res = f"DefaultEndpointsProtocol={matched.group(2)};AccountName={matched.group(4)};AccountKey=***;EndpointSuffix={matched.group(8)}"
+            return res
+        except Exception as e:
+            logger.warning(
+                f"Something went wrong when trying to obfuscate URL for Azure: {e}"
+            )
+
```

Comment:
I think here we should just raise the error instead of catching in a try except.

Reply:
I agree with you on this one. I was trying to preserve the existing pattern, but think that this is our chance to "make things better". Update coming up